server/src/main/java/org/apache/iotdb/db/engine/snapshot/SnapshotTaker.java

```diff
@@ -117,83 +208,152 @@ public boolean takeFullSnapshot(String snapshotDirPath, boolean flushBeforeSnaps
     } finally {
       manager.readUnlock();
     }
+    return true;
+  }
+
+  private boolean createSnapshotInAnotherDisk(File snapshotDir) {
+    for (String dataDir : IoTDBDescriptor.getInstance().getConfig().getDataDirs()) {
+      File seqDir =
+          new File(
+              dataDir,
+              "sequence"
+                  + File.separator
+                  + dataRegion.getLogicalStorageGroupName()
+                  + File.separator
+                  + dataRegion.getDataRegionId());
+      File unseqDir =
+          new File(
+              dataDir,
+              "unsequence"
+                  + File.separator
+                  + dataRegion.getLogicalStorageGroupName()
+                  + File.separator
+                  + dataRegion.getDataRegionId());
+      File localSeqSnapshotDir =
+          new File(
+              dataDir,
+              "snapshot"
+                  + File.separator
+                  + "sequence"
+                  + File.separator
+                  + snapshotDir.getName()
+                  + File.separator
+                  + dataRegion.getLogicalStorageGroupName()
+                  + File.separator
+                  + dataRegion.getDataRegionId());
+      File localUnseqSnapshotDir =
+          new File(
+              dataDir,
+              "snapshot"
+                  + File.separator
+                  + "unsequence"
+                  + File.separator
+                  + snapshotDir.getName()
+                  + File.separator
+                  + dataRegion.getLogicalStorageGroupName()
+                  + File.separator
+                  + dataRegion.getDataRegionId());
+      if (!localSeqSnapshotDir.mkdirs()) {
+        LOGGER.warn("Failed to create local snapshot dir {}", localSeqSnapshotDir);
+        return false;
+      }
+      List<Long> timePartitions = dataRegion.getTimePartitions();
+      TsFileManager manager = dataRegion.getTsFileManager();
+      manager.readLock();
+      for (long timePartition : timePartitions) {
+        File seqTimePartitionDir = new File(seqDir, String.valueOf(timePartition));
+        File seqTimePartitionSnapshotDir =
+            new File(localSeqSnapshotDir, String.valueOf(timePartition));
 
-    LOGGER.info(
-        "Successfully take snapshot for {}-{}, snapshot directory is {}",
-        dataRegion.getLogicalStorageGroupName(),
-        dataRegion.getDataRegionId(),
-        snapshotDirPath);
+        try {
+          createFileSnapshotToTargetOne(seqTimePartitionDir, seqTimePartitionSnapshotDir);
+        } catch (IOException e) {
+          LOGGER.error(
+              "Failed to create snapshot for {}-{}",
+              dataRegion.getLogicalStorageGroupName(),
+              dataRegion.getDataRegionId(),
+              e);
+        } finally {
+          cleanUpWhenFail(snapshotDir);
```

Comment:
cleanUpWhenFail should be called in catch clause instead of final clause?

Reply:
This finally is unnecessary, so I've deleted it.